superset/views/core.py

```diff
@@ -959,7 +959,7 @@ def save_or_overwrite_slice(
             slc = Slice(owners=[g.user] if g.user else [])
 
         form_data["adhoc_filters"] = self.remove_extra_filters(
-            form_data.get("adhoc_filters", [])
+            (form_data.get("adhoc_filters", []) or [])
```

Comment:
nit (I thought I added this comment, but it seems to have disappeared into the ether); I think this should work just as well to handle where `adhoc_filters` is set to `None`:
```suggestion
            form_data.get("adhoc_filters") or []
```

Reply:
I don't really remember why I did that :sweat_smile:
But agree :+1: